ebos/eclwriter.hh

```diff
@@ -208,6 +209,27 @@ public:
         if (totalCpuTime != 0.0) {
             miscSummaryData["TCPU"] = totalCpuTime;
         }
+        if (this->sub_step_report_.total_newton_iterations != 0) {
+            miscSummaryData["NEWTON"] = this->sub_step_report_.total_newton_iterations;
+        }
+        if (this->sub_step_report_.total_linear_iterations != 0) {
+            miscSummaryData["MLINEARS"] = this->sub_step_report_.total_linear_iterations;
+        }
+        if (this->sub_step_report_.total_newton_iterations != 0) {
+            miscSummaryData["NLINEARS"] =  this->sub_step_report_.total_linear_iterations / this->sub_step_report_.total_newton_iterations;
```

Comment:
This is an integer.  Did you consider making it a floating-point value instead?

Reply:
It was supposed to be floating-point, but you are right its not. I though it should work since 
`std::map<std::string, double> miscSummaryData;` 
I guess I need use `static_cast`

Reply:
That was better. Now NLINEARS is indeed the average. Thanks for noticing.

Reply:
> I though it should work since `std::map<std::string, double> miscSummaryData;`

You're right that the value will be **stored** as a floating-point value, but one of the pitfalls of C++ is that the expression
```
integer / integer
```
performs integer division.  This means that the value of that expression will be an integer too and that that integer will be converted to a floating-point value when it's stored in the `map<>`.

> I guess I need use `static_cast`.

Yes, that's the simplest workaround.